packages/core/src/lib/list-box/__tests__/list-box.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { describe, it, expect } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { ListBox } from '$lib';
+import { default as ListBoxSlot } from '../list-box-slot.svelte';
+
+const defaultProps = {
+  disabled: false,
+  left: 'leftText1,leftText2,leftText3',
+  right: 'rightText1,rightText2,rightText3',
+  leftlabel: 'Left',
+  rightlabel: 'Right',
+  height: '200px',
+  suffix: false,
+};
+
+describe('ListBox', () => {
+  it('Renders left options and can move them to the right', async () => {
+    const user = userEvent.setup();
+    render(ListBoxSlot, { ...defaultProps });
+
+    await user.click(screen.getByText('leftText1'));
+    await user.click(screen.getByText('leftText2'));
+    await user.click(screen.getByText('leftText3'));
+
+    const moveRightButton = screen.getByTestId('move-right');
+    await user.click(moveRightButton);
+
+    expect(screen.getByText('leftText1')).toBeVisible();
```

Comment:
When possible, put elements in a variable so you're only calling `getByWhatever` once. The only time to re-call `getBy` for the same element is if it's getting mounted / unmounted during the course of the test

Reply:
Im not a 100% sure if this is how it works but screen.get returns an html element. I was un sure if that element is a ref to the value on the screen or if it is just a copy of the html element. So to be safe I called it before the elements get moved around and after to be safe. Do you know which case is true? ref to the element or copy of the element?

Reply:
In `testing-library`, `getByXyz` returns the actual DOM element. If the implementation of the component creates new elements rather than moving them, then yes, you will need to call it again. In that case, I would still recommend assigning to a variable for readability and to communicate the intent of your tests

```ts
const goodNameForThisElement = screen.getByXyz(...)

expect(goodNameForThisElement).toBeVisible()
```

(Playwright is different, in that its methods return `Locators`, which are not evaluated until you call `expect`. I kind of prefer this approach! But testing-library's approach is certainly more simple)